### PR TITLE
Fix - Modify regex for Safari

### PR DESCRIPTION
**Pull request by halocline**
[Deploy Preview](https://deploy-preview-2056--keen-mayer-a86c8b.netlify.app/)

#### What does this PR do?

Closes https://github.com/grommet/hpe-design-system/issues/2053

#### Where should the reviewer start?

Search.js

#### What testing has been done on this PR?

Manual testing on Safari, Chrome, and Firefox.

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [x] Small, medium, and large screen sizes
- [x] Cross-browsers (FireFox, Chrome, and Safari)
- [x] Light & dark modes
- [x] All hyperlinks route properly

**Accessibility Checks**
- [x] Keyboard interactions
- [x] Screen reader experience
- [x] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [x] Console is free of warnings and errors
- [x] Passes E2E commit checks
- [x] Visual snapshots are reasonable

#### How should this be manually tested?

Use the site search functionality on Safari, Chrome, Firefox.

#### What are the relevant issues?

https://github.com/grommet/hpe-design-system/issues/2053

### CODE REVIEW

File: aries-site/src/layouts/navigation/Search.js [hunk 1]
```diff
@@ -201,10 +201,11 @@ export const Search = ({ setOpen }) => {
     let id;
     if (matches && matches.length > 0) {
       const { index, input } = matches[0];
-      const regexp = new RegExp(/#{1,} (...+?)(?<= ~{2})/, 'g');
+      const regexp = new RegExp(/#{1,} (...+?) ?~{2}/, 'g');
       const headings = [...input.matchAll(regexp)];
       // Find nearest preceding heading to query match
       headings.forEach(heading => {
+        console.log(heading);
```
Comment (taysea) [inline aries-site/src/layouts/navigation/Search.js:208]: Should we remove this?
```suggestion
```

Comment (halocline) [reply aries-site/src/layouts/navigation/Search.js]: Done.

**Review by jcfilben — COMMENTED** [review]
Do we also need to remove `?<=` in the `scripts/site-contents.mjs` file at line 46?

**Comment by halocline** [comment]
> Do we also need to remove `?<=` in the `scripts/site-contents.mjs` file at line 46?

Good question. I don't believe so since that code is run by a Node process on the server prior to the project build and not the browser.

**Review by jcfilben — APPROVED** [review]
Looks good!

**Pull request merged**